### PR TITLE
Improve contributing file

### LINKED ISSUE #2: Reaktoro's workflow
Hi @allanleal,

Since Reaktoro is growing, I think it is a great opportunity to start thinking about the workflow that could be used inside the project. After we came into an agreement about that topic, I could create an official file and add to the project.

What do you think about the idea of using a Fork & Pull Request workflow? After we decided to work in a new Feature or fix a bug, we could create a new branch from our master fork, add all the code in that branch and, once we finish, make a PR to the reaktoro's main branch. 

Doing that, we will only have one branch inside the repo and the releases could me done by the flag system offer by github.

It is a simpler version of gitflow, because won't have any developer, but I think that could work fine. 

[]'s

### PR DESCRIPTION
**Pull request by alexandrebbruno**
Hey guys, 
I'm opening this PR just to improve documentation by adding some information about our workflow, as discussed at #2 

fix #2

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 
-    enter your email here
+1. Fork this repository
+2. Clone the repository at your machine
+3. Add your changes in a branch named after what's being done (`lower-case-with-hyphens`)
+4. Make a pull request to `reaktoro/Reaktoro`, targeting the `master` branch
+
+Before starting any change to Reaktoro, we would love to hear about and discuss your idea in a GitHub issue. In this way, we can help ensure that your time will be well spent and your contribution will be accepted.
+
+If you are interested in contributing with the development of Reaktoro but don't know how, please get in touch.
```
Comment (tadeu) [inline CONTRIBUTING.md]: We should probably add a way to get in touch... an e-mail?

**Comment by tadeu** [comment]
@nicoddemus, sorry, I was editing via GH interface at the same time you reviewed, it seems your review got "lost" (`tadeu dismissed nicoddemus’s stale review via 5451772`)

**Comment by nicoddemus** [comment]
No worries, I will review this again tomorrow morning! 👍

File: CONTRIBUTING.md (hunk 1 above)
Comment (nicoddemus) [reply CONTRIBUTING.md]: I suggest to use issues instead, so change this to "please create an issue."

Comment (tadeu) [reply CONTRIBUTING.md]: Even for things like "I'm interested in contributing but don't know how"?
Hmm, it makes sense, documentation probably should be updated if someone comes with a problem like this :)

Comment (alexandrebbruno) [reply CONTRIBUTING.md]: I considered create an email for the Reaktoro's organisation to put here, but I think that will add another communication channel that needs to be checked. Waiting that the user will find us in a social media it isn't a create option either. 😝
So, I think that the best option is tell than to create an issue, like @nicoddemus said 👍

**Comment by allanleal** [comment]
@alexandrebbruno please read [here](https://github.com/reaktoro/Reaktoro/pull/6/commits/b6a1d73efa16e0dd27e270935e0d16127d1a4dea) my proposed version of the Contributing page. It is a bit more comprehensive. Thanks a lot for starting this.

**Comment by alexandrebbruno** [comment]
Hey @allanleal,
great job!
Talking about the idea of adding a subsection explaining how to make an questions. Don´t you think that would be better if put that information in de README.mb? Since it is an important topic, make it more visible at the README.mb could be better 🤔

**Comment by allanleal** [comment]
Yes, that's what we should do it (if all agree)! I think we should have 
a very nice README file, with all important information in there. 
Non-critical stuff could go to different pages.

File: README.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ Reaktoro is a unified framework for modeling chemically reactive systems. It pro
 
 For installation instructions, tutorials, and list of publications related to this project, please access [reaktoro.org](http://www.reaktoro.org). This web site describes how to download and install Reaktoro, and demonstrate some basic usage.
 
+## Contributing
+
```
[the remark this answers is not in the document]
Comment (nicoddemus) [reply README.md]: In my experience the README file should be targeted toward users, so adding the entire contribution text in there will make the README larger than needs to be and add noise to the majority of users. Contributors by definition are willing to explore both the codebase and documentation more thoroughly, in order to see where they can contribute and how, so they won't miss that information by accident.

**Comment by nicoddemus** [comment]
> Talking about the idea of adding a subsection explaining how to make an questions. Don´t you think that would be better if put that information in de README.mb?

GitHub has [issue templates](https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/) which is also a great way to communicate this.

**Comment by allanleal** [comment]
I agree with your view on this.

How about then we at least have in the README the following sections 
(maybe subsections in a FAQ section?!):
/
//"How to report bugs", //
//"How to ask questions"//
//"How can I share my results"/

And in the Contributing page, we have:

/"How can I help with documentation"//
//"How can I help with code development"/

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (allanleal) [reply README.md]: I agree with your view on this.

How about then we at least have in the README the following sections (maybe subsections in a FAQ section?!):

**_How to report bugs
How to ask questions
How can I share my results_**

And in the Contributing page, we have:

**_How can I help with documentation
How can I help with code development_**

[the remark this answers is not in the document]
Comment (nicoddemus) [reply README.md]: Sounds good! If we keep each topic brief, with one or two paragraphs, it won't increase the size of the readme too much. 😁

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,7 +1,59 @@
 Contributing
```
Comment (nicoddemus) [inline CONTRIBUTING.md]: Please convert this file to `.rst` for consistency. 👍

Comment (alexandrebbruno) [reply CONTRIBUTING.md]: working on it :)
- [  ] converte CONTRIBUTING.md to .rst

**Comment by allanleal** [comment]
Shouldn't we favor Markdown syntax instead of reStructuredText? The former is a lot simpler, albeit with certain limitations (which might not be critical here). Are there any strong reasons why we should go for the latter?

**Comment by nicoddemus** [comment]
> Shouldn't we favor Markdown syntax instead of reStructuredText? The former is a lot simpler, albeit with certain limitations (which might not be critical here).

Personally I prefer reStructuredText because we can easily include it in our Sphinx documentation (when we have one). Also because `Markdown` lacks a standard, it is tricky to use it for documentation. See [this post](http://ericholscher.com/blog/2016/mar/15/dont-use-markdown-for-technical-docs/) by Eric Holster (one of the founders of ReadTheDocs).

Having said that if you prefer Markdown for the files in the root directory we can switch it back easily with [pandoc](https://pandoc.org/). 😁 

> Another thing. What about relying on word wrapping functionality from text editors (with configurable line width) instead of entering our own line breaks in the documentation?

I don't have a strong opinion here and I'm OK with whichever way you guys decide! 👍

**Comment by allanleal** [comment]
I have seen those arguments already. They make sense, also the lack of
standard. My experience with markdown is overall positive, except when I
want to customize things, and then I need to enter HTML directly. Having
said that, let's continue then with rst.

However, I would propose the docs to be written without inserting line
breaks to conform with some predefined maximum line width. We can just rely
on the editor with word wrapping mode enabled.

**Comment by nicoddemus** [comment]
> However, I would propose the docs to be written without inserting line breaks to conform with some predefined maximum line width. We can just rely on the editor with word wrapping mode enabled.

Fine by me! 👍 We should add a paragraph stating this practice in the "Contributing with documentation and examples" section.

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+Get in touch with us by going to `Reaktoro's GitHub Issues <https://github.com/reaktoro/Reaktoro/issues/new>`__ and telling us what you want to share.
+
+Contributing with documentation and examples
+--------------------------------------------
+
+You have used Reaktoro for a while and wants to contribute with some examples on how to use it for solving some specific problems. Thank you for your interest. We appreciate your effort and willingness to contribute.
+
+Please go to `Reaktoro's GitHub Issues <https://github.com/reaktoro/Reaktoro/issues/new>`__ and write a new issue, detailing what you want to do. Please select the label **enhancement** on the right side of the issues page.
+
+If you start to write any documentaiton and examples, please make sure to do it without inserting line breaks to conform with some predefined maximum line width.
```
Comment (nicoddemus) [inline CONTRIBUTING.rst]: The last part of this phrase is a little confusing, I suggest to change to:

```rst
.. note:: 
    When writing documentation, do not insert line breaks within paragraphs (configure your editor to wrap lines automatically instead if you prefer).
```

**Comment by nicoddemus** [comment]
🎉

**Comment by allanleal** [comment]
Great, thanks! :)

**Comment by allanleal** [comment]
Just noticed that the link to `Contributing.rst` is not correct (on the last sentence of the README file). I don't know how to get it correct using `rst`.

**Comment by nicoddemus** [comment]
> Just noticed that the link to Contributing.rst is not correct (on the last sentence of the FAQ in the README file). I don't know how to get it correct using rst. 

Good catch, fixed in e1a4894c

**Pull request merged**